File: js/programmers/level1/K번째수.js
function solution(array, commands) {
  const answer = [];
  for (let command of commands) {
    answer.push(
      array.slice(command[0] - 1, command[1]).sort((a, b) => a - b)[
        command[2] - 1
      ],
    );
  }
  return answer;
}

console.log(
  solution(
    [1, 5, 2, 6, 3, 7, 4],
    [
      [2, 5, 3],
      [4, 4, 1],
      [1, 7, 3],
    ],
  ),
);

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] result = new int[commands.length];
        
        for (int i=0; i<commands.length; i++) {
            int a = commands[i][0];
            int b = commands[i][1];
            int c = commands[i][2];
            
            int[] newArr = Arrays.copyOfRange(array, a-1, b);
            Arrays.sort(newArr);
            
            result[i] = newArr[c-1];
        }
        
        return result;
    }
}